src/main/java/com/pickyfy/pickyfy/domain/PlaceCategory.java

```diff
@@ -0,0 +1,36 @@
+package com.pickyfy.pickyfy.domain;
+
+import jakarta.persistence.Entity;
+import jakarta.persistence.FetchType;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.JoinColumn;
+import jakarta.persistence.ManyToOne;
+import lombok.AccessLevel;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+
+@Entity
+@Getter
+@NoArgsConstructor(access = AccessLevel.PROTECTED)
+public class PlaceCategory {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @ManyToOne(fetch = FetchType.LAZY)
+    @JoinColumn(name = "place_id", nullable = false)
+    private Place place;
+
+    @ManyToOne(fetch = FetchType.LAZY)
+    @JoinColumn(name = "category_id", nullable = false)
+    private Category category;
+
+    @Builder
+    public PlaceCategory(Place place, Category category) {
+        this.place = place;
+        this.category = category;
+    }
+}
```

Comment:
```suggestion
@Entity
@Getter
@Builder
@AllArgsConstructor
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class PlaceCategory {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "place_id", nullable = false)
    private Place place;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "category_id", nullable = false)
    private Category category;
}
```
생성자를 직접 작성하여 @Builder를 사용하기보다 @AllArgsConstructor와 @Builder를 class단에 작성하는게 더 간결해 좋을 것 같습니다.
다른 클래스들도 변경 부탁드립니다

Reply:
오 좋은 의견이네요!! 근데 저는 이 부분에 대해서 의견이 좀 다른데요..! `@AllArgsConstructor`은 지양해야 하지 않나라고 생각합니다. 그 이유는

1. 접근할 필요없는 필드(id 등)에 접근할 수 있다는 부분이 꺼려짐
- 물론 id는 자동생성되도록 설정이 되어있기 때문에 무시되겠지만 객체 생성 시 @AllArgsConstructor로 id 필드에 대한 접근 가능성을 열어놓는 것 자체가 선호되는 방식은 아닌 듯합니다.

2. 같은 타입의 매개변수 순서 바뀔 때 문제 감지 불가
- 생성자를 생성할 때 같은 타입의 매개변수가 존재하는데 이 순서를 바꿔서 넣어도 정상적으로 생성되어서 오류 감지가 어렵기 때문에 문제 발생 가능성이 있을 것 같습니다.

----

이렇게 @AllArgsConstructor를 지양하자는 여러 이유들이 있긴 하지만 객체를 생성할 때 Builder 패턴만 사용하도록 하자!라는 규약을 우리끼리 정하고 그것을 잘 준수한다면 문제는 없을 것 같긴 합니다.
만약 다들 편리성을 더 추구하신다면 위와 같이 변경해도 큰 문제는 없을 것 같습니다!

아니면 `@AllArgsConstructor(access = AccessLevel.PRIVATE)`으로 하면 문제가 없을 것 같은데 이런식으로도 사용을 하는지는 잘 모르겠네요.

Reply:
말씀하신 대로 @AllArgsConstructor를 사용하면 불필요하게 특정 필드에 접근 가능성을 열어두는 문제가 있을 수 있다는 점, 같은 타입의 매개변수 순서 변경에 따른 오류 감지가 어렵다는 점에 대해 동의합니다

추가로 만약 생성자를 직접 명시해서 사용하더라도 2번 문제는 발생할 수 있으니 접근제어자를 private으로 변경해야할 거같습니다

Reply:
제가 생각했을 땐 builder 패턴으로 작성하자고 해서, 굳이 @AllArgsConstructor를 사용 해야하나 생각합니다. builder 패턴으로 객체 생성시 가독성이 좀 더 좋기도 하고, 가변성제어가 용이하기 때문에 저는 @AllArgsConstructor 를 지양해야하지 않을까 라는 생각입니다.

Reply:
접근 제어자는 private으로 했어야 하는데 제 실수입니다..ㅎㅎ

나머지는 효빈님이 적어주신 생각과 비슷하고요, @AllArgsConstructor를 사용하면 모든 맴버 변수를 builder로 생성해야 하기 때문에 원하는 속성만 생성메소드(builder)로 넣어 생성할 수 없게 됩니다. 물론 이 부분은 DTO가 해결해 주긴 하지만 필요 없는 기능은 제약하는 것이 좋다고 생각합니다.

Reply:
@AllArgsConstructor를 사용하면 모든 맴버 변수를 builder로 생성해야 하기 때문에 원하는 속성만 생성메소드(builder)로 넣어 생성할 수 없게 됩니다. 

혹시 이 부분 더 자세히 설명해주실 수 있나요?

Reply:
위에 댓글 보니 아마 이해하신 것 같은데 제가 말을 혼동되게 썼네요. 조금 더 찾아본 내용 공부하자는 차원에서 적어둘게요!

@AllArgsConstructor를 사용하면 모든 필드를 파라미터로 받는 생성자가 만들어지고, @Builder가 빌드된 객체를 생성할 때 해당 생성자를 호출하게 됩니다.

반면에 생성자를 직접 작성하면 필수 필드를 명확히 지정할 수 있어 필요한 필드만 선택적으로 빌더에 노출할 수 있습니다.

또한, 초기화 로직을 명시적으로 제어할 수 있습니다. 우리 프로젝트로 예시로 SavedPlace의 isPublic을 초기 생성자에서 True로 세팅하여 항상 Public으로 시작하도록 비즈니스 규칙을 생성자에서 처리할 수 있습니다.

Reply:
전 기본적으로 엔티티를 작성할 때
@Builder
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor
이 3개는 기본으로 설정하고 만들었는데, 좋은 고민인 것 같습니다.

Builder 패턴의 경우에는 AllArgsConstructor와는 별개로 선택적으로 인자를 받아 객체를 생성할 수 있다는 이점과, setter 메소드로 값을 넣어주기 보다 한 번에 하나의 완전한 객체로 만들 수 있다는 장점 때문에 사용하지 않나 생각합니다.

@RequiredArgsConstructor 를 사용해보는 것도 괜찮은 방법이라고 생각합니다.


[롬복 @All/NoArgsConstructor 제대로 알고 사용해보자.](https://velog.io/@code-10/%EB%A1%AC%EB%B3%B5-AllNoArgsConstructor-%EC%A0%9C%EB%8C%80%EB%A1%9C-%EC%95%8C%EA%B3%A0-%EC%82%AC%EC%9A%A9%ED%95%B4%EB%B3%B4%EC%9E%90)